Content.Client/Sprite/ContentSpriteSystem.cs

```diff
@@ -0,0 +1,200 @@
+using System.IO;
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using Content.Client.Administration.Managers;
+using Content.Shared.Database;
+using Content.Shared.Verbs;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface;
+using Robust.Shared.ContentPack;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+using Color = Robust.Shared.Maths.Color;
+
+namespace Content.Client.Sprite;
+
+public sealed class ContentSpriteSystem : EntitySystem
+{
+    [Dependency] private readonly IClientAdminManager _adminManager = default!;
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IResourceManager _resManager = default!;
+    [Dependency] private readonly IUserInterfaceManager _ui = default!;
+
+    private ContentSpriteControl _control = new();
+
+    public static readonly ResPath Exports = new ResPath("/Exports");
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _resManager.UserData.CreateDir(Exports);
+        _ui.RootControl.AddChild(_control);
+        SubscribeLocalEvent<GetVerbsEvent<Verb>>(GetVerbs);
+    }
+
+    public override void Shutdown()
+    {
+        base.Shutdown();
+
+        foreach (var queued in _control._queuedTextures)
+        {
+            queued.Tcs.SetCanceled();
+        }
+
+        _control._queuedTextures.Clear();
+
+        _ui.RootControl.RemoveChild(_control);
+    }
+
+    /// <summary>
+    /// Exports sprites for all directions
+    /// </summary>
+    public async Task Export(EntityUid entity, CancellationToken cancelToken = default)
+    {
+        var tasks = new Task[4];
+        var i = 0;
+
+        foreach (var dir in new Direction[]
+                 {
+                     Direction.South,
+                     Direction.East,
+                     Direction.North,
+                     Direction.West,
+                 })
+        {
+            tasks[i++] = Export(entity, dir, cancelToken);
+        }
+
+        await Task.WhenAll(tasks);
+    }
+
+    /// <summary>
+    /// Exports the sprite for a particular direction.
+    /// </summary>
+    public async Task Export(EntityUid entity, Direction direction, CancellationToken cancelToken = default)
+    {
+        if (!_timing.IsFirstTimePredicted)
+            return;
+
+        if (!TryComp(entity, out SpriteComponent? spriteComp))
+            return;
+
+        // Don't want to wait for engine pr
+        var size = Vector2i.Zero;
+
+        foreach (var layer in spriteComp.AllLayers)
+        {
+            if (!layer.Visible)
+                continue;
+
+            size = Vector2i.ComponentMax(size, layer.PixelSize);
+        }
+
```

Comment:
Will probably just engine pr this ideally the pr gets reviewed but what are the odds of that haha